streams/src/main/java/org/apache/kafka/streams/state/WindowStore.java

```diff
@@ -57,6 +57,10 @@
 
     /**
      * Put a key-value pair into the window with given window start timestamp
+     * <p>
+     * If serialized value bytes are null it is treated as delete. Note that as an optimization, deletes will be
+     * skipped in the case of an underlying store that allows duplicates.
```

Comment:
I would use the 'interpreted' instead of 'treated' to be more consistent with the rest of the javadoc (at least in this file). The behavior is not really due to an optimization and that's not really the point IMO. Also, 'retain duplicates' in the more common term instead of 'allow duplicates'. I would rephrase it to something like this:
```suggestion
     * If serialized value bytes are null it is treated as delete. Note that deletes will be
     * skipped in the case of an underlying store that retains duplicates.
```

Reply:
lol, I forgot to apply my own suggestion to use 'interpreted', good catch.